src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs

```diff
@@ -257,10 +257,7 @@ private static bool ContainsUpperCaseLetter(string pattern)
             // only after we've tried all other forms first.  This is the weakest of all matches.  For example, if
             // user types 'bin' we want to match 'OperatorBinary' (start of word) or 'BinaryInformationNode' (camel
             // humps) before matching 'Combine'.
-            // 
-            // We only do this for strings longer than three characters to avoid too many false positives when the
-            // user has only barely started writing a word.
```

Comment:
i'm a bit concerned here that the entire reason why we did this is going away :)

Reply:
Could we just change this to two letters.  That way a single letter doesn't just match randomly in all things (especially common when a peron just types one char to bring up completion), while two characters are maybe enough to feel like it's ok to find substring matches.

Reply:
Typing one character will bring up completion regardless, and as I mentioned in the issue these results would be far down and won't make much of a difference unless we filter the completion results.

![image](https://github.com/dotnet/roslyn/assets/46457277/9aa1f1a7-641c-49b8-83ce-44625c29fd35)

But yeah, I somewhat agree with your point on having at least two characters.
@sharwell since you triaged this, do you have a strong opinion about this?